Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,196 @@
+            Logger.browserLogger.error("LeanplumIntegration - Could not load settings from Info.plist")
+            return
+        }
+
+        switch settings.environment {
+        case .development:
+            Logger.browserLogger.info("LeanplumIntegration - Setting up for Development")
+            Leanplum.setDeviceId(ASIdentifierManager.shared().advertisingIdentifier.uuidString)
+            Leanplum.setAppId(settings.appId, withDevelopmentKey: settings.key)
+        case .production:
+            Logger.browserLogger.info("LeanplumIntegration - Setting up for Production")
+            Leanplum.setAppId(settings.appId, withProductionKey: settings.key)
+        }
+        Leanplum.syncResourcesAsync(true)
+        setupTemplateDictionary()
+
+        var userAttributesDict = [AnyHashable: Any]()
+        userAttributesDict["Focus Installed"] = "false"
+        userAttributesDict["Klar Installed"] = "false"
+        userAttributesDict["Alternate Mail Client Installed"] = "mailto:"
+
+        if let focusURL = URL(string: "firefox-focus://"), UIApplication.shared.canOpenURL(focusURL) {
+            userAttributesDict["Focus Installed"] = "true"
+        }
+
+        if let klarURL = URL(string: "firefox-klar://"), UIApplication.shared.canOpenURL(klarURL) {
+            userAttributesDict["Klar Installed"] = "true"
+        }
+
+        if let mailtoScheme = self.profile?.prefs.stringForKey(PrefsKeys.KeyMailToOption), mailtoScheme != "mailto:" {
+            userAttributesDict["Alternate Mail Client Installed"] = mailtoScheme
+        }
+
+        Leanplum.start(userAttributes: userAttributesDict)
+
+        Leanplum.track(LeanplumEventName.openedApp.rawValue)
+    }
+
+    // Events
+
+    func track(eventName: LeanplumEventName) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue)
+        }
+    }
+
+    func track(eventName: LeanplumEventName, withParameters parameters: [String: AnyObject]) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue, withParameters: parameters)
+        }
+    }
+
+    // States
+
+    func advanceTo(state: String) {
+        if shouldSendToLP() {
+            Leanplum.advance(to: state)
+        }
+    }
+
+    // Data Modeling
+
+    func setupTemplateDictionary() {
+        if shouldSendToLP() {
+            LPVar.define("Template Dictionary", with: ["Template Text": "", "Button Text": "", "Deep Link": "", "Hex Color String": ""])
+        }
+    }
+
+    func getTemplateDictionary() -> [String:String]? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: ["Template Dictionary"]) as? [String : String]
+        }
+        return nil
+    }
+
+    func getBoolVariableFromServer(key: String) -> Bool? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: [key]) as? Bool
+        }
+        return nil
+    }
+
+    // Utils
+    
+    func setEnabled(_ enabled: Bool) {
```

Comment:
Nit: Use properties instead of Java style setters.

```swift
var enabled: Bool {
    didSet: {
          if enabled {
                 start()
          }
    }
}
```